x-pack/plugins/monitoring/public/alerts/components/param_details_form/expression.tsx

```diff
@@ -92,11 +92,13 @@ export const Expression: React.FC<Props> = (props) => {
 
   const onFilterChange = useCallback(
     (filter: string) => {
-      setRuleParams('filterQueryText', filter);
-      setRuleParams(
-        'filterQuery',
-        convertKueryToElasticSearchQuery(filter, derivedIndexPattern) || ''
-      );
+      if (derivedIndexPattern) {
```

Comment:
I have no idea if this is a good idea but the types work. Did this a couple of times

Reply:
Under what circumstances might it fail to create the data view? 
I feel like if we cannot render the kuery bar here that's not something we should silently fail on.

And a nitpick, can we turn these into guard statements instead?

Reply:
The only way it would fail to load a data view is if the `getFieldsForWildcard` call within `dataViews.create` fails - which is to say that this will fail in the same circumstances but might have different consequences.

Reply:
@miltonhultgren I implemented your request for guard statements. Please verify that I understood you correctly.

There are two particular places I wanted to rely on someone familiar with monitoring - this specific conditional AND the rendering of the KueryBar. To be blunt, I'm not quite sure what I'm doing here 😬 but wanted to take a swing at it. Worst case scenario - your team loops back to this when you have time. Best case scenario - I highlighted a small technicality that can be accounted for with little effort.